openerp/addons/base/res/res_bank.py

```diff
@@ -95,8 +102,44 @@ def _default_value(self, cursor, user, field, context=None):
                 value = address.get(field, value)
         return value
 
+    def _get_sanitized_account_number(
+            self, cr, uid, ids, field_name, arg, context):
+        """Compute sanitized version off account number.
+
+        This is needed for reliable search and uniqueness test."""
+        res = {}
+        for this_obj in self.browse(cr, uid, ids, context = context):
+            res[this_obj.id] = (
+                sanitize_account_number(this_obj.acc_number))
+        return res
+
+    def search(self, cr, user, args, offset=0, limit=None, order=None,
+               context=None, count=False):
+        pos = 0
+        while pos < len(args):
+            if args[pos][0] == 'acc_number':
+                op = args[pos][1]
+                value = args[pos][2]
+                if hasattr(value, '__iter__'):
+                    value = [sanitize_account_number(i) for i in value]
+                else:
+                    value = sanitize_account_number(value)
+                if 'like' in op:
+                    value = '%' + value + '%'
+                args[pos] = ('sanitized_acc_number', op, value)
+            pos += 1
+        return super(res_partner_bank, self).search(
+            cr, user, args, offset=0, limit=None, order=None, context=None,
+            count=False)
+
     _columns = {
         'acc_number': fields.char('Account Number', required=True),
+        'sanitized_acc_number': fields.function(
+            _get_sanitized_account_number,
+            string='Sanitized Account Number',
+            type='char', size=64, readonly=True,
+            store=True,
+        ),
         'bank': fields.many2one('res.bank', 'Bank'),
```

Comment:
Maybe we could push the refactoring as to clean res.partner.bank model. Things like account.bank.bic / account.bank_bic and account.bank.name / account.bank_name duplication or having the account type called state and not being required are bothersome